### ISSUE TITLE
Suppport ScreenCaptureKit loopback

### ISSUE THREAD

**Issue by thewh1teagle** [RustAudio/cpal#876]
In recent versions of `macOS` there's new API called [screen capture kit](https://developer.apple.com/documentation/screencapturekit/) which provides a way to capture the screen content (Audio / Video),
That way we can receive audio from the speakers, similar how `cpal` does that with WSAPI on Windows (Added in #251)
As starting point we can follow the approach used in the project [scap](https://github.com/helmerapp/scap) in [engine/mac/mod.rs](https://github.com/helmerapp/scap/blob/main/scap/src/capturer/engine/mac/mod.rs) they capture audio / video using `screen capture kit`

**Comment by LatentDream** [comment]
+1

**Comment by rustdesk** [comment]
+1

**Comment by rustdesk** [comment]
Maybe, we can add a virtual device called loopback, which use screencapturekit, the others keep no change

**Comment by louis030195** [comment]
+1

**Comment by SuperKenVery** [comment]
ScreenCaptureKit has a relatively high latency, in my previous experience. Maybe we could have a look at [CoreAudio Tap API](https://developer.apple.com/documentation/coreaudio/audiohardwarecreateprocesstap(_:_:)?language=objc)

[Example usage](https://gist.github.com/sudara/34f00efad69a7e8ceafa078ea0f76f6f)

**Comment by SuperKenVery** [comment]
Hey guys I implemented loopback recording on macOS with CoreAudio Tap API. 

Here's my pr: #1003

**Comment by thewh1teagle** [comment]
Hey @SuperKenVery! I tested it on macOS 26.3 (apple chip) but unfortunately it doesn't work (it's recording silence)

**Comment by thewh1teagle** [comment]
I will just use this fork https://github.com/rustdesk-org/cpal/tree/osx-screencapturekit

**Comment by SuperKenVery** [comment]
@thewh1teagle Probably permission issue: https://github.com/RustAudio/cpal/pull/894#issuecomment-3823323664 (see last section)

**Comment by thewh1teagle** [comment]
Thank you @SuperKenVery! you were right it recorded silence just because of missing permission. 
I made this small crate https://github.com/thewh1teagle/vibe/tree/main/crates/macos-permissions so I can ask/check permission and it works great now

**Comment by thewh1teagle** [comment]
actually on prod build I get this error first time I allow it

```console
The requested device is no longer available. For example, it has been unplugged.

Location:
    desktop/src-tauri/src/cmd/audio.rs:215:18
```

**Comment by SuperKenVery** [comment]
@thewh1teagle Super cool, would you be interested in merging this into cpal?

**Comment by thewh1teagle** [comment]
@SuperKenVery sure! created PR for fixing the bug + permission helpers

**Comment by louis030195** [comment]
@thewh1teagle btw i was wondering if you found better way to capture speaker audio than sck? users keep complaining that it shows the Apple thing in the tray bar "recording screen" although it only records mic (as seen when using Notion recorder or Granola too)

**Comment by thewh1teagle** [comment]
@louis030195 yup, exactly what I did yesterday it doesn't show the 'screen recording' icon
see the latest code of https://github.com/thewh1teagle/vibe I just created a fork of cpal with support for system audio recording + permissions

**Comment by offbyseven** [comment]
@louis030195 not sure if you found a solution here, but because screencapturekit accesses the screen, macos legally has to show that orange microphone icon in the tray bar when the user is recording the screen. Like others are saying, core audio taps is a better fit if you are just trying to record audio - since it’s audio-only, it doesn’t require the screen-recording permission and doesn’t need that icon in the tray bar.
